src/box/lua/config/applier/box_status.lua

```diff
@@ -0,0 +1,56 @@
+local mkversion = require('internal.mkversion')
+
+local box_status_state = {
+    config = nil,
+    watcher = nil,
+}
+
+local SCHEMA_VERSION_ALERT_KEY = 'schema_version_is_outdated'
+
+local function set_schema_version_alert(current_version, latest_version)
+    box_status_state.config._aboard:set({
+        type = 'warn',
+        message = (
+            'The schema version %s is outdated, the latest ' ..
+            'version is %s. Please, consider using box.schema.upgrade().')
+            :format(current_version, latest_version),
+    }, {
+        key = SCHEMA_VERSION_ALERT_KEY,
+    })
+end
+
+local function drop_schema_version_alert()
+    box_status_state.config._aboard:drop(SCHEMA_VERSION_ALERT_KEY)
+end
+
+local function check_schema_version()
+    local current_version = mkversion.get()
+    local latest_version = mkversion.get_latest()
+    if current_version < latest_version then
+        set_schema_version_alert(current_version, latest_version)
+    else
+        drop_schema_version_alert()
+    end
+end
+
+local function box_status_watcher()
+    return box.watch('box.status', function(key)
+        assert(key == 'box.status')
+        check_schema_version()
+    end)
+end
+
+local function apply(config)
+    if box_status_state.watcher ~= nil then
+        box_status_state.watcher:unregister()
+    end
```

Comment:
How about registering it on demand? Re-registering each time seems redundant.

Reply:
I rewrote the `apply` method, now I don't store the config itself, requiring it when I need access to the aboard itself only. And register the watcher only one time (the first time the `apply` is called.